editor.planx.uk/src/pages/Pay/MakePayment.tsx

```diff
@@ -44,137 +50,189 @@ enum PaymentState {
 export default function MakePayment({
   sessionPreviewData,
   createdAt,
-  paymentRequestId,
+  id: paymentRequestId,
+  govPayPaymentId,
   paymentAmount,
+  paidAt,
 }: PaymentRequest) {
+  const { address, rawProjectTypes } =
+    parseSessionPreviewData(sessionPreviewData);
   const [currentState, setState] = useState<
     (typeof States)[keyof typeof States]
   >(States.Init);
-  const [loading, isLoading] = useState(true);
-  const [payment, setPayment] = useState(
-    sessionPreviewData.govUkPayment || undefined
-  );
+  const [isLoading, setIsLoading] = useState(true);
+  const [payment, setPayment] = useState<GovUKPayment | undefined>(undefined);
+  const flowName = useStore((state) => state.flowName);
+  const theme = useTheme();
 
+  // Pass async errors up to ErrorBoundary
+  const [errorMessage, setErrorMessage] = useState<string | undefined>();
```

Comment:
Errors thrown inside an async function (e.g. `updatePaymentState()`) don't get caught by the `ErrorBoundary` so we need a bit of manual handling for this here.